### PR TITLE
core: introduce parse_ip_port 

### PR DESCRIPTION
**Pull request by ssahani**
1. Listed in TODO.
2. Tree wide replace safe_atou16 with parse_ip_port incase
   it's used for ports.

### CODE REVIEW

File: src/basic/parse-util.c [hunk 1]
```diff
@@ -574,3 +574,29 @@ int parse_nice(const char *p, int *ret) {
...
+                return -EINVAL;
+        if (s[0] == '-')
+                return -ERANGE;
+        if ((unsigned long) (uint16_t) l != l)
+                return -ERANGE;
+        if (l == 0)
+                return -EINVAL;
+
+        *ret = (uint16_t) l;
+        return 0;
```
Comment (poettering) [inline src/basic/parse-util.c:591]: why not reuse safe_atou16() for this and then make an additional check for != 0?

**Comment by ssahani** [comment]
Actually thinking about it thanks.

**Comment by ssahani** [comment]
updated .

**Comment by poettering** [comment]
thanks! looks great! thanks for working on this!

**Pull request merged**